pages/common/perl.md

```diff
@@ -30,3 +30,7 @@
 - Run a multi-line find/replace expression on a file, and save the result in another file:
 
 `perl -p0e 's/{{foo\nbar}}/{{foobar}}/g' {{input_file}} > {{output_file}}`
+
+- Run a regular expression on stdin, printing out first capture group for each line:
+
+`echo "$INPUT" | perl -nle 'if (/.*({{foo}}).*/) {print "$1"; last;}'`
```

Comment:
Can we use something more basic than `echo "$INPUT"`? E.g. `cat {{somefile.txt}}` or something else along those lines.

Also, it's not immediately clear to me what specific feature this example intends to demonstrate. Is it the `$1`? Or maybe the `{...}` construct? The `/.* ... .*/` pattern? The `-n` option, or the `-l` one?

We should be careful not to overload a single example with multiple new concepts. Examples should work as building blocks that users can compose, not necessarily ready-to-use "incantations" that serve a particular purpose but fail to teach the user about the mechanics of the command.